internal/trace/trace.go

```diff
@@ -0,0 +1,60 @@
+package trace
+
+import (
+	"context"
+	"os"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+func MaybeTrace(serviceVersion string) bool {
+	if os.Getenv("UPM_TRACE") != "1" {
+		return false
+	}
+
+	replid := os.Getenv("REPL_ID")
+	if replid == "" {
+		return false
+	}
+
+	tracer.Start(
+		tracer.WithService("upm"),
+		tracer.WithGlobalTag("replid", replid),
+		tracer.WithServiceVersion(serviceVersion),
+	)
+	return true
```

Comment:
We should unset the DD_TRACE_ID and DD_SPAN_ID from the process environment if we can here. That way it doesn't pollute the environment processes upm spawns, like package managers, unless we explicitly want to profile those at a later date and we handle it then.

Then below we'll have to get the traceID and spanID from memory instead of the environment.

Reply:
:+1: